store/sqlstore/team_store.go

```diff
@@ -256,6 +270,7 @@ func (s SqlTeamStore) GetByInviteId(inviteId string) store.StoreChannel {
 
 		if err := s.GetReplica().SelectOne(&team, "SELECT * FROM Teams WHERE Id = :InviteId OR InviteId = :InviteId", map[string]interface{}{"InviteId": inviteId}); err != nil {
 			result.Err = model.NewAppError("SqlTeamStore.GetByInviteId", "store.sql_team.get_by_invite_id.finding.app_error", nil, "inviteId="+inviteId+", "+err.Error(), http.StatusNotFound)
+			return
```

Comment:
Here is the change that generate the change in the tests. before this change the error was set to the result.Err but the code keep running, and then reach the other error in the function, and finally set the data (to nil, because the result of the requests was nil because can't find the team with that invite id).